twitter/direct_messages_test.go

```diff
@@ -0,0 +1,169 @@
+		assertMethod(t, "GET", r)
+		assertQuery(t, map[string]string{"id": testDMIDStr}, r)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, `[`+testDMJSON+`]`)
+	})
+
+	client := NewClient(httpClient)
+	dms, _, err := client.DirectMessages.Show(testDMID)
+	expected := []DirectMessage{testDM}
+	assert.Nil(t, err)
+	assert.Equal(t, expected, dms)
+}
+
+func TestDirectMessageService_Get(t *testing.T) {
+	httpClient, mux, server := testServer()
+	defer server.Close()
+
+	mux.HandleFunc("/1.1/direct_messages.json", func(w http.ResponseWriter, r *http.Request) {
+		assertMethod(t, "GET", r)
+		assertQuery(t, map[string]string{"since_id": "589147592367431680", "count": "1"}, r)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, `[`+testDMJSON+`]`)
+	})
+
+	client := NewClient(httpClient)
+	params := &DirectMessageGetParams{SinceID: 589147592367431680, Count: 1}
+	dms, _, err := client.DirectMessages.Get(params)
+	expected := []DirectMessage{testDM}
+	assert.Nil(t, err)
+	assert.Equal(t, expected, dms)
+}
+
+func TestDirectMessageService_GetNilParams(t *testing.T) {
+	httpClient, mux, server := testServer()
+	defer server.Close()
+
+	mux.HandleFunc("/1.1/direct_messages.json", func(w http.ResponseWriter, r *http.Request) {
+		assertMethod(t, "GET", r)
+		assertQuery(t, map[string]string{}, r)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, `[`+testDMJSON+`]`)
+	})
+
+	client := NewClient(httpClient)
+	dms, _, err := client.DirectMessages.Get(nil)
+	expected := []DirectMessage{testDM}
+	assert.Nil(t, err)
+	assert.Equal(t, expected, dms)
+}
+
+func TestDirectMessageService_Sent(t *testing.T) {
+	httpClient, mux, server := testServer()
+	defer server.Close()
+
+	mux.HandleFunc("/1.1/direct_messages/sent.json", func(w http.ResponseWriter, r *http.Request) {
+		assertMethod(t, "GET", r)
+		assertQuery(t, map[string]string{"since_id": "589147592367431680", "count": "1"}, r)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, `[`+testDMJSON+`]`)
+	})
+
+	client := NewClient(httpClient)
+	params := &DirectMessageSentParams{SinceID: 589147592367431680, Count: 1}
+	dms, _, err := client.DirectMessages.Sent(params)
+	expected := []DirectMessage{testDM}
+	assert.Nil(t, err)
+	assert.Equal(t, expected, dms)
+}
+
+func TestDirectMessageService_SentNilParams(t *testing.T) {
+	httpClient, mux, server := testServer()
+	defer server.Close()
+
+	mux.HandleFunc("/1.1/direct_messages/sent.json", func(w http.ResponseWriter, r *http.Request) {
+		assertMethod(t, "GET", r)
+		assertQuery(t, map[string]string{}, r)
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, `[`+testDMJSON+`]`)
+	})
+
+	client := NewClient(httpClient)
+	dms, _, err := client.DirectMessages.Sent(nil)
+	expected := []DirectMessage{testDM}
+	assert.Nil(t, err)
+	assert.Equal(t, expected, dms)
+}
+
+func TestDirectMessageService_SendToID(t *testing.T) {
```

Comment:
Removing the SendToID and SendToScreenName test shows that the underlying endpoint `New` isn't tested, which is disguised by the way gocover super smart. I favor removing `SendToID` and `SendToScreenName` convenience methods and just testing `New`.

If you're looking for an example of where a similar decision was made, check TimelineService.UserTimeline. It similarly requires either a user id or a user screen name. In such cases, in my opinion, it is simplest to expose the parameters and leave writing convenience wrappers to users who want it.